### PR TITLE
Crowdloan Refactoring for XCM helper pallet

### PR DESCRIPTION
**Pull request by ashi20192020ayush**

### CODE REVIEW

File: pallets/parallel-xcm/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type XCM: ParallelXCM;
+    }
+
```
Comment (ETeissonniere) [inline pallets/parallel-xcm/src/lib.rs:32]: The `parallel-xcm` exposes the `ParallelXCM` trait. The code you currently have would be for the other pallets that uses this one.

File: pallets/crowdloans/src/lib.rs [hunk 2]
```diff
@@ -504,10 +489,7 @@ pub mod pallet {
                 amount,
                 false,
             )?;
-            TotalReserves::<T>::try_mutate(|b| -> DispatchResult {
-                *b = b.checked_add(amount).ok_or(ArithmeticError::Overflow)?;
-                Ok(())
-            })?;
+            T::XCM::update_total_reserves(amount)?;
```
Comment (MrPai) [inline pallets/crowdloans/src/lib.rs:492]: 1/3
`update_total_reserves` invoked in crowdloan pallet, and also the transferring happens here

File: pallets/parallel-xcm/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    fn update_xcm_fees(fees: BalanceOf<T>) {
+        XcmFees::<T>::mutate(|v| *v = fees);
+    }
+
+    fn update_total_reserves(reserves: BalanceOf<T>) -> DispatchResult {
+        TotalReserves::<T>::try_mutate(|b| -> DispatchResult {
+            *b = b.checked_add(reserves).ok_or(ArithmeticError::Overflow)?;
+            Ok(())
+        })
+    }
```
Comment (MrPai) [inline pallets/parallel-xcm/src/lib.rs:132]: 2/3 `TotalReserves` updated here, however, the asset actually transferred outside.

File: pallets/parallel-xcm/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        let asset: MultiAsset = (MultiLocation::here(), fees).into();
+
+        match xcm_fees_payment_strategy {
+            XcmFeesPaymentStrategy::Reserves => {
+                T::Assets::burn_from(relay_currency, &account_id, fees)?;
+
+                TotalReserves::<T>::try_mutate(|b| -> DispatchResult {
+                    *b = b.checked_sub(fees).ok_or(ArithmeticError::Underflow)?;
+                    Ok(())
+                })?;
```
Comment (MrPai) [inline pallets/parallel-xcm/src/lib.rs:153]: 3/3 now decrease `TotalReserves` here, and do asset transferring inside this pallet, 
if it is a bug?

Comment (MrPai) [reply pallets/parallel-xcm/src/lib.rs:153]: Maybe here `account_id` should always be PalletId account, but I suggest that we can move this code inside into `T::XCM::update_total_reserves`
https://github.com/parallel-finance/parallel/blob/2d5382a471e2a0ce0b04579af662f37830d23fb9/pallets/crowdloans/src/lib.rs#L500-L506

File: pallets/crowdloans/src/lib.rs (hunk 2 above)
Comment (ashi20192020ayush) [reply pallets/crowdloans/src/lib.rs:492]: @MrPai Are you saying that Asset transfer should also happen inside `update_total_reserves`?

File: pallets/parallel-xcm/src/lib.rs (hunk 3 above)
Comment (ashi20192020ayush) [reply pallets/parallel-xcm/src/lib.rs:132]: same as above

File: pallets/parallel-xcm/src/lib.rs (hunk 4 above)
Comment (ashi20192020ayush) [reply pallets/parallel-xcm/src/lib.rs:153]: @MrPai Regarding these comments, Please create separate issue. As a part of this PR, I just moved XCM related functions from Crowdloan to XCM Helper. 

For ex: I moved entire this methid https://github.com/parallel-finance/parallel/blob/master/pallets/crowdloans/src/lib.rs#L657 

cc @ETeissonniere

File: pallets/crowdloans/src/lib.rs (hunk 2 above)
Comment (MrPai) [reply pallets/crowdloans/src/lib.rs:526]: yes, that's what I mean

File: pallets/parallel-xcm/src/lib.rs (hunk 4 above)
Comment (MrPai) [reply pallets/parallel-xcm/src/lib.rs:153]: yes, please close this issue #1002 once finished

Comment (ashi20192020ayush) [reply pallets/parallel-xcm/src/lib.rs:153]: @MrPai I am marking this comment as resolved. #1002 would be done separately. and this is not related to this PR. 

cc @ETeissonniere

**Review by MrPai — APPROVED** [review]
looks good for me

**Pull request merged**